tests/Symfony/SymfonyContainerResultCacheMetaExtensionTest.php

```diff
@@ -0,0 +1,402 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Symfony;
+
+use PHPStan\Testing\PHPStanTestCase;
+use function count;
+use function file_put_contents;
+use function sys_get_temp_dir;
+use function tempnam;
+use function var_export;
+
+final class SymfonyContainerResultCacheMetaExtensionTest extends PHPStanTestCase
+{
+
+	private static string $configFilePath;
+
+	/**
+	 * This test has to check if hash of the Symfony Container is correctly calculated,
+	 * in order to do that we need to dynamically create a temporary configuration file
+	 * because `PHPStanTestCase::getContainer()` caches container under key calculated from
+	 * additional config files' paths, so we can't reuse the same config file between tests.
+	 */
+	public static function getAdditionalConfigFiles(): array
+	{
+		return [
+			__DIR__ . '/../../extension.neon',
+			self::$configFilePath,
```

Comment:
I don't understand the need for a static property here. There are always problems with them.

Reply:
Please have a look [here](#discussion_r1922158510) for details.